src/Entity/EntityId.php

```diff
@@ -18,6 +18,13 @@
 	protected $serialization;
 
 	/**
+	 * @param $serialization
+	 */
+	protected function __construct( $serialization ) {
+		$this->serialization = $serialization;
```

Comment:
Should fail if $serialization[0] === ':', and also if $serialization is empty or has the wrong type. Keep the checks simple though, this is a potential performance hotspot, and callers (subclass constructors) should already be doing normalization.